File: src/backend/src/main/java/com/personal/finance/backend/notifications/service/NotificationService.java
package com.personal.finance.backend.notifications.service;

import com.personal.finance.backend.notifications.dto.response.NotificationDTO;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface NotificationService {
    Page<NotificationDTO> getMyNotifications(Long userId, Pageable pageable);
    void markAsRead(Long id, Long userId);
    void markAllAsRead(Long userId);
    void deleteNotification(Long id, Long userId);

    void createSystemNotification(Long userId, String title, String content);
}

File: src/frontend/src/services/notificationService.ts
import api from './api';

const notificationService = {
  getMyNotifications: async (params?: { page: number, size: number }) => {
    const response = await api.get('/notifications', { params });
    return response.data;
  },

  markAsRead: async (id: number) => {
    await api.patch(`/notifications/${id}/read`);
  },

  markAllAsRead: async () => {
    await api.patch('/notifications/read-all');
  },

  deleteNotification: async (id: number) => {
    await api.delete(`/notifications/${id}`);
  }
};

export default notificationService;